charmcraft/extensions/extension.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2018-2023 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Extension base class definition."""
+
+import abc
+import os
+import sys
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Sequence, Tuple, final
+
+from craft_cli import emit
+
+from charmcraft import errors
+
+
+class Extension(abc.ABC):
+    """Extension is the class from which all extensions inherit.
+
+    Extensions have the ability to add snippets to apps, parts, and indeed add new parts
```

Comment:
```suggestion
    Extensions have the ability to add snippets to charms, parts, and indeed add new parts
```

Reply:
we should just remove apps, it refers to the apps entry in snapcraft.yaml; perhaps replaced with config, actions, ....